### PR TITLE
You can now shoot OVER candles to light them

### PR DESCRIPTION
**Pull request by Repth1**
[tweak]
title a.k.a candle powercreep

so it turns out Crossed() is a thing, thanks @Exxion 
an extension to #21448 
this also allows emitters to light candles, and firing directly at candles functions as before

fun image

:cl:
 * tweak: You can now shoot over candles with lasers or emitters to light them.

### CODE REVIEW

File: code/game/objects/items/candle.dm [hunk 1]
```diff
@@ -76,3 +72,13 @@
 	if(lit)
 		return source_temperature
 	return 0
+
+/obj/item/candle/Crossed(var/obj/Proj)
+	if(istype(Proj, /obj/item/projectile/beam))
```
Comment (jknpj) [inline code/game/objects/items/candle.dm]: Make an isBeam() macro instead of doing this manually twice

Comment (Repth1) [reply code/game/objects/items/candle.dm]: Sure, where would it live though?

**Comment by Kitfox88** [comment]
>you will never light a cigar in your mouth by shooting it with a laser but shoot yourself in the face accidentally

File: code/game/objects/items/candle.dm (hunk 1 above)
Comment (Repth1) [reply code/game/objects/items/candle.dm]: ignore me i'm stupid

Comment (Repth1) [reply code/game/objects/items/candle.dm]: done

**Comment by Repth1** [comment]
> > you will never light a cigar in your mouth by shooting it with a laser but shoot yourself in the face accidentally

#21578

File: code/game/objects/items/candle.dm [hunk 2]
```diff
@@ -76,3 +72,10 @@
 	if(lit)
 		return source_temperature
 	return 0
+
+/obj/item/candle/Crossed(var/obj/Proj)
+	if(isbeam(Proj))
+		//Can also be an obj/effect, but they both have the damage var
```
Comment (jknpj) [inline code/game/objects/items/candle.dm:78]: This is some oldcoder-tier hack. You might need to use if(Proj?:damage && Proj?:damage !=0) instead. ?: is one of lummox's new null-conditional operators, but i'm not sure how haram is to use what is pretty much a new runtime operator.

Comment (Repth1) [reply code/game/objects/items/candle.dm:78]: i can separate them out to an isbeam and isbeameffect function, but i feel that would put us back where we started

Comment (jknpj) [reply code/game/objects/items/candle.dm:78]: Damian said it's fine to keep it as it is, since isbeam() is filtering the object.

Comment (Exxion) [reply code/game/objects/items/candle.dm:78]: @jknpj I don't think those operators do what you think they do
This is "fine"
By which I mean it really isn't but there isn't a better alternative thanks to BYOND

**Pull request merged**